### PR TITLE
chore: add latest pallets [#14][#17]

### LINKED ISSUE #14: pallet-murmur: implement create wallet + proxy extrinsics

### LINKED ISSUE #17: Misc: Add various pallets to pallets repo
- murmur pallet
- modified proxy pallet
- modified scheduler pallet

### PR DESCRIPTION
**Pull request by driemworks**
- adds the latest version of the proxy, scheduler, randomness beacon, and murmur pallets
- closes #14 
- closes #17

### CODE REVIEW

**Review by colemanirby — APPROVED** [review]

File: client/consensus/beefy-etf/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+sc-network-sync = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
+sc-network-types = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
+sc-utils = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
+sc-transaction-pool-api = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
+sp-api = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
+sp-application-crypto = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
+sp-arithmetic = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
+sp-blockchain = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
+sp-consensus = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
+# sp-consensus-beefy-etf-etf = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
```
Comment (juangirini) [inline client/consensus/beefy-etf/Cargo.toml:39]:
```suggestion
```

File: client/consensus/beefy-etf/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+sp-consensus-beefy-etf = { path = "../../../primitives/consensus/beefy-etf", features = ["bls-experimental"] }
+sp-core = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
+sp-crypto-hashing = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing" }
+sp-keystore = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", features = ["bls-experimental"]}
+# sp-keystore = { package = "sp-keystore-etf",path = "../../../primitives/keystore", features = ["bls-experimental"]}
```
Comment (juangirini) [inline client/consensus/beefy-etf/Cargo.toml:44]:
```suggestion
```

File: pallets/beefy-etf/Cargo.toml [hunk 3]
```diff
@@ -21,7 +21,8 @@ frame-system = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch
 pallet-authorship = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 pallet-etf = { path = "../etf", default-features = false}
 pallet-session = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
-sp-consensus-beefy-etf = { git = "https://github.com/ideal-lab5/etf.git", default-features = false, features = ["serde", "bls-experimental"] }
+# sp-consensus-beefy-etf = { git = "https://github.com/ideal-lab5/etf.git", branch = "tony/dev", default-features = false, features = ["serde", "bls-experimental"] }
```
Comment (juangirini) [inline pallets/beefy-etf/Cargo.toml:24]:
```suggestion
```

File: pallets/beefy-mmr-etf/Cargo.toml [hunk 4]
```diff
@@ -23,7 +23,8 @@ frame-system = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch
 pallet-beefy = { package = "pallet-beefy-etf", path = "../beefy-etf", default-features = false }
 pallet-mmr = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 pallet-session = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
-sp-consensus-beefy-etf = { git = "https://github.com/ideal-lab5/etf.git", default-features = false, features = ["bls-experimental"]}
+# sp-consensus-beefy-etf = { git = "https://github.com/ideal-lab5/etf.git", branch = "tony/dev", default-features = false, features = ["serde", "bls-experimental"] }
```
Comment (juangirini) [inline pallets/beefy-mmr-etf/Cargo.toml:26]:
```suggestion
```

File: pallets/etf/Cargo.toml [hunk 5]
```diff
@@ -20,7 +20,8 @@ frame-support = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch
 frame-system = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 pallet-authorship = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 pallet-session = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
-sp-consensus-beefy-etf = { git = "https://github.com/ideal-lab5/etf.git", default-features = false, features = ["serde", "bls-experimental"] }
+# sp-consensus-beefy-etf = { git = "https://github.com/ideal-lab5/etf.git", branch = "tony/dev", default-features = false, features = ["serde", "bls-experimental"] }
```
Comment (juangirini) [inline pallets/etf/Cargo.toml:23]:
```suggestion
```

File: pallets/murmur/Cargo.toml [hunk 6]
```diff
@@ -7,7 +7,7 @@ homepage = "https://idealabs.network"
 edition = "2021"
 license = "GPL-3.0-or-later WITH Classpath-exception-2.0"
 publish = false
-repository.workspace = true
+repository = "https://github.com/ideal-lab5/etf/"
```
Comment (juangirini) [inline pallets/murmur/Cargo.toml:10]:
```suggestion
repository = "https://github.com/ideal-lab5/pallets/"
```
Thought we should rename the repo...

File: pallets/murmur/Cargo.toml [hunk 7]
```diff
@@ -30,29 +30,41 @@ frame-system = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch
 sp-std = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 sp-runtime = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 sp-core = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
-# etf-crypto-primitives = { default-features =  false, git = "https://github.com/ideal-lab5/etf-sdk.git", branch = "tony/dev"}
 ckb-merkle-mountain-range = { version = "0.5.2", default-features = false }
 sha3 = { version = "0.10.8", default-features = false }
 murmur-core = { package="murmur-core", git = "https://github.com/ideal-lab5/murmur.git", branch = "dev", default-features = false }
-# totp-rs = { version = "5.5.1", default-features = false, optional = true }
+# murmur-core = { package = "murmur-core", path = "../../../murmur/core", default-features = false }
```
Comment (juangirini) [inline pallets/murmur/Cargo.toml:36]:
```suggestion
```

File: pallets/murmur/Cargo.toml [hunk 8]
```diff
@@ -30,29 +30,41 @@ frame-system = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch
...
 frame-support-test = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+sp-staking = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 pallet-balances = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 pallet-timestamp = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+pallet-session = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+pallet-mmr = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+pallet-etf = { path = "../etf", default-features = false }
+pallet-beefy-etf = { path = "../beefy-etf", default-features = false }
+pallet-beefy-mmr-etf = { path = "../beefy-mmr-etf", default-features = false }
+# sp-consensus-beefy-etf = { git = "https://github.com/ideal-lab5/etf.git", branch = "tony/dev", features = ["bls-experimental"] }
```
Comment (juangirini) [inline pallets/murmur/Cargo.toml:52]:
```suggestion
```

File: pallets/murmur/Cargo.toml [hunk 9]
```diff
@@ -30,29 +30,41 @@ frame-system = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch
...
+sp-state-machine = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 rand_chacha ="0.3.1"
 ark-std = { version = "0.4.0", default-features = false }
 hex = "0.4.3"
 sp-io = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false  }
+etf-crypto-primitives = { default-features =  false, git = "https://github.com/ideal-lab5/etf-sdk.git", branch = "dev"}
+w3f-bls = { version = "0.1.3", default-features = false }
+binary-merkle-tree = { version = "15.0.0", default-features = false }
+murmur-test-utils = { git = "https://github.com/ideal-lab5/murmur.git", branch = "dev" }
+# murmur-test-utils = { path = "../../../murmur/test-utils" }
```
Comment (juangirini) [inline pallets/murmur/Cargo.toml:63]:
```suggestion
```

File: pallets/murmur/src/tests.rs [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+
+#[test]
+fn it_can_create_new_proxy_with_unique_name() {
+	let seed = b"seed".to_vec();
+	let unique_name = b"name".to_vec();
+	let bounded_name = BoundedVec::<u8, ConstU32<32>>::truncate_from(unique_name);
+	let block_schedule = vec![1, 2, 3];
+	let ephem_msk = [3; 32];
+
+	// let root = vec![1,2,3];
```
Comment (juangirini) [inline pallets/murmur/src/tests.rs:64]:
```suggestion
```

File: pallets/murmur/src/tests.rs [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+			bounded_name.clone(),
+		));
+
+		// check storage
+		let registered_proxy = murmur::Registry::<Test>::get(bounded_name.clone());
+		assert!(registered_proxy.is_some());
+		// 	// verify data
+		// 	assert_eq!(registered_proxy.root, root);
+		// 	assert_eq!(registered_proxy.size, size);
+		// 	assert_eq!(registered_proxy.name, unique_name);
```
Comment (juangirini) [inline pallets/murmur/src/tests.rs:91]:
```suggestion
```

File: pallets/randomness-beacon/Cargo.toml [hunk 12]
```diff
@@ -15,16 +15,16 @@ workspace = true
...
 serde = { version = "1.0.197", default-features = false }
 
 frame-support = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 frame-system = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 pallet-session = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
 pallet-authorship = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
-sp-consensus-beefy-etf = { git = "https://github.com/ideal-lab5/etf.git", default-features = false, features = ["serde", "bls-experimental"] }
-pallet-etf = { path = "../etf", default-features = false}
-pallet-beefy = { package = "pallet-beefy-etf", path = "../beefy-etf", default-features = false}
+# sp-consensus-beefy-etf = { git = "https://github.com/ideal-lab5/etf.git", branch = "tony/dev", default-features = false, features = ["serde", "bls-experimental"] }
```
Comment (juangirini) [inline pallets/randomness-beacon/Cargo.toml:24]:
```suggestion
```

File: primitives/consensus/beefy-etf/Cargo.toml [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+serde = { optional = true, features = ["alloc", "derive"], workspace = true }
+sp-api = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+sp-application-crypto = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+sp-core = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+sp-crypto-hashing = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+sp-io = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+sp-mmr-primitives = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+sp-runtime = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+sp-keystore = { git = "https://github.com/ideal-lab5/polkadot-sdk.git", branch = "testing", default-features = false }
+# sp-keystore = { package = "sp-keystore-etf", path = "../../keystore-etf", default-features = false }
```
Comment (juangirini) [inline primitives/consensus/beefy-etf/Cargo.toml:29]:
```suggestion
```

File: primitives/consensus/beefy-etf/src/test_utils.rs [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+// 	MsgHash: Hash,
+// 	<MsgHash as Hash>::Output: Into<[u8; 32]>,
+// {
+// 	fn sign_with_hasher(&self, message: &[u8]) -> <Self as AppCrypto>::Signature {
+// 		let hashed_message = <MsgHash as Hash>::hash(message).into();
+// 		self.as_inner_ref().sign_prehashed(&hashed_message).into()
+// 	}
+// }
+
+// #[cfg(feature = "bls-experimental")]
```
Comment (juangirini) [inline primitives/consensus/beefy-etf/src/test_utils.rs:66]:
```suggestion
```

**Review by juangirini — APPROVED** [review]
LGTM. It can be merged after cleaning up the comments.
Also, unrelated to this PR, we should rename this repo as it is not only pallets anymore

**Pull request merged**